code/game/gamemodes/miniantags/bot_swarm/swarmer.dm

```diff
@@ -242,8 +242,12 @@
 	return TRUE
 
 /obj/machinery/door/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
+	for(var/turf/T in range(1, src))
+		if(isspaceturf(T) || istype(T.loc, /area/space))
+			S << "<span class='warning'>Destroying this object has the potential to cause a hull breach. Aborting.</span>"
+			S.target = null
+			return TRUE
 	S.DisIntegrate(src)
-	return TRUE
```

Comment:
No. This should be returning true so AI swarmers see the airlock as a valid object if they succeed in eating it.